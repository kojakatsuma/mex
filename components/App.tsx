import { Posts } from './mdx';
import { OGPHeader } from './OGPHeader';
import Link from 'next/link';

const App = () => {
  return (
    <>
      <OGPHeader url={'/log'} title='log' metaDescription={'日記です'} />
      {Posts.map((Post, i) => {
        return (
          <div key={i}>
            <Post />
            <Link href='/'>
              <h3 className='menu'>{'back to top'}</h3>
            </Link>
            <hr />
          </div>
        );
      })}
    </>
  );
};
export default App;
