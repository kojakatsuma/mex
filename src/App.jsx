import React, { useEffect, useState } from 'react';
import { Container, Typography, Paper, Divider, createMuiTheme, ThemeProvider, CssBaseline, Fade } from '@material-ui/core';
import { Posts } from './mdx';
import FontFaceObserver from 'fontfaceobserver';
import 'github-markdown-css'
import './index.css'

import MPLUSRounded1c from './fonts/MPLUSRounded1c-Light.ttf';


const mPlusRoundedlc = {
  fontFamily: 'MPLUSRounded1c',
  fontDisplay: 'swap',
  src: `
  local('MPLUSRounded1c'), url(${MPLUSRounded1c}) format('truetype');
  `
}

const colorCode = '#ffffff'


const theme = createMuiTheme({
  palette: {
    background: {
      paper: colorCode,
      default: colorCode
    }
  },
  typography: {
    fontFamily: "MPLUSRounded1c,sans-serif",
  }, overrides: {
    MuiCssBaseline: {
      '@global': {
        '@font-face': [mPlusRoundedlc],
        backgroundColor: 'white'
      },
    }
  }
})

export default () => {
  const [loaded, setLoaded] = useState(false)
  useEffect(() => {
    const font = new FontFaceObserver('MPLUSRounded1c')
    font.load().then(() => {
      setLoaded(true)
    })
  })
  return (
    <ThemeProvider theme={theme}>
      <Container maxWidth={false} style={{maxWidth: 680}}>
        <CssBaseline />
        {loaded ? <Fade in={loaded} timeout={{ enter: 800 }}>
          <Paper elevation={0}>
            <Typography variant="h2" >mex</Typography>
            <Typography variant="subtitle2" paragraph>here is mexico.</Typography>
            {
              Posts.map((Post, i) => {
                return (
                  <React.Fragment key={i}>
                    <Divider />
                    <Paper style={{ marginBottom: '2em' }} square elevation={0} className='markdown-body' >
                      <Typography component='article' >
                        <Post />
                      </Typography>
                    </Paper>
                  </React.Fragment>
                )
              })
            }
          </Paper>
        </Fade> : <div>loading</div>
        }
      </Container>
    </ThemeProvider>
  )
}
