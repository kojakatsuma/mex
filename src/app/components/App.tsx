import Header from './Header';
import { Posts } from './mdx';
import { Typography, Divider } from '@material-ui/core';

const App = () => (
  <main>
    <Header />
    {
      Posts.map((Post, i) => {
        return (
          <Typography component='article' key={i}>
            <Post />
            <Divider />
          </Typography>
        )
      })
    }
  </main>
)

export default App
