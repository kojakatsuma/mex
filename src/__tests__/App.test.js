import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';


test('App test', () => {
  const tree = renderer.create(<App/>).toJSON()
  expect(tree).toMatchSnapshot();
})
