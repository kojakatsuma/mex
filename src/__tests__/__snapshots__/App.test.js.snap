// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthMd"
>
  <h1
    className="MuiTypography-root MuiTypography-h1"
  >
    mex
  </h1>
  <p
    className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-paragraph"
  >
    here is mexico.
  </p>
  <hr
    className="MuiDivider-root"
  />
  <div
    className="MuiPaper-root markdown-body MuiPaper-elevation0"
    style={
      Object {
        "marginBottom": "2em",
      }
    }
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      <20200130-2.mdx />
    </p>
  </div>
  <hr
    className="MuiDivider-root"
  />
  <div
    className="MuiPaper-root markdown-body MuiPaper-elevation0"
    style={
      Object {
        "marginBottom": "2em",
      }
    }
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      <20200130.mdx />
    </p>
  </div>
  <hr
    className="MuiDivider-root"
  />
  <div
    className="MuiPaper-root markdown-body MuiPaper-elevation0"
    style={
      Object {
        "marginBottom": "2em",
      }
    }
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      <20200129.mdx />
    </p>
  </div>
</div>
`;
